src/test/regress/expected/multi_insert_select.out

```diff
@@ -765,10 +762,16 @@ DEBUG:  distributed statement: INSERT INTO multi_insert_select.agg_events_133000
    raw_events_second.user_id
  FROM
    reference_table LEFT JOIN raw_events_second ON reference_table.user_id = raw_events_second.user_id;
-DEBUG:  distributed statement: INSERT INTO multi_insert_select.agg_events_13300008 AS citus_table_alias (user_id) SELECT raw_events_second.user_id FROM (multi_insert_select.reference_table_13300012 reference_table LEFT JOIN multi_insert_select.raw_events_second_13300004 raw_events_second ON ((reference_table.user_id OPERATOR(pg_catalog.=) raw_events_second.user_id))) WHERE (raw_events_second.user_id IS NOT NULL)
-DEBUG:  distributed statement: INSERT INTO multi_insert_select.agg_events_13300009 AS citus_table_alias (user_id) SELECT raw_events_second.user_id FROM (multi_insert_select.reference_table_13300012 reference_table LEFT JOIN multi_insert_select.raw_events_second_13300005 raw_events_second ON ((reference_table.user_id OPERATOR(pg_catalog.=) raw_events_second.user_id))) WHERE (raw_events_second.user_id IS NOT NULL)
-DEBUG:  distributed statement: INSERT INTO multi_insert_select.agg_events_13300010 AS citus_table_alias (user_id) SELECT raw_events_second.user_id FROM (multi_insert_select.reference_table_13300012 reference_table LEFT JOIN multi_insert_select.raw_events_second_13300006 raw_events_second ON ((reference_table.user_id OPERATOR(pg_catalog.=) raw_events_second.user_id))) WHERE (raw_events_second.user_id IS NOT NULL)
-DEBUG:  distributed statement: INSERT INTO multi_insert_select.agg_events_13300011 AS citus_table_alias (user_id) SELECT raw_events_second.user_id FROM (multi_insert_select.reference_table_13300012 reference_table LEFT JOIN multi_insert_select.raw_events_second_13300007 raw_events_second ON ((reference_table.user_id OPERATOR(pg_catalog.=) raw_events_second.user_id))) WHERE (raw_events_second.user_id IS NOT NULL)
+DEBUG:  cannot perform a lateral outer join when a distributed subquery references a reference table
```

Comment:
When throwing an error in DeferredErrorIfUnsupportedRecurringTuplesJoin, we assume that the outer join is a lateral one because recursive planner should've already handled the such a join already, see https://github.com/citusdata/citus/pull/6512. However, this is not the case for INSERT .. SELECT. Hence the debug message is misleading.

Also copy pasting the relevant code-comment from DeferredErrorIfUnsupportedRecurringTuplesJoin:
```c
		 * For outer joins, this can only happen if it's a lateral outer join
		 * where the inner distributed subquery references the recurring outer
		 * rel. This because, such outer joins should not appear here because
		 * the recursive planner (RecursivelyPlanRecurringTupleOuterJoinWalker)
		 * should have already planned the non-recurring side if it wasn't a
		 * lateral join. For this reason, if the outer join is between a recurring
		 * rel --on the outer side-- and a non-recurring rel --on the other side--,
		 * we throw an error assuming that it's a lateral outer join. ...
```

Reply:
Besides that, this is one of the cases where we were incorrectly pushing down some insert .. select queries on main (and not on that branch).